wallet-new/docs/wallet-tls.md

```diff
@@ -0,0 +1,146 @@
+<div id="table-of-contents">
+<h2>Table of Contents</h2>
+<div id="text-table-of-contents">
+<ul>
+<li><a href="#sec-1">1. Introduction</a></li>
+<li><a href="#sec-2">2. Abstract</a></li>
+<li><a href="#sec-3">3. Motivation</a></li>
+<li><a href="#sec-4">4. Current design</a>
+<li><a href="#sec-5">5. Proposed design</a>
+<ul>
+<li><a href="#sec-5-1">5.1. Nodejs frontend change</a></li>
+<li><a href="#sec-5-2">5.2. Haskell backend change</a></li>
+<ul>
+</li>
+</ul>
+</div>
+</div>
+
+# Introduction<a id="sec-1" name="sec-1"></a>
+
+# Abstract<a id="sec-2" name="sec-2"></a>
+
+The purpose of this document is to review security model in communication
+ between wallet backend and wallet frontend (Daedalus).
+
+# Motivation<a id="sec-3" name="sec-3"></a>
+
+Communication between wallet frontend (Daedalus) and wallet backend in cardano
+ is done via http server. There are various attacks possible to this kind of 
+environmnet (TODO footnote or further list of possible attack vectors needed).
+
+# Current design<a id="sec-4" name="sec-4"></a>
+
+This summarises current approach in server-client communication.
+
+-   wallet server has https server (based on TLS CA server cert) set up and listening on specific IP:PORT
+    
+-   client is using its on pair of CA certificate in order to verify server cert when sending request to server
```

Comment:
_its own pair_, maybe?

Reply:
Where is the evidence (with hypelinks to actual code) to back those assertions up? I guess that's arguably the most useful aspect of a technical proposal: bridge the gaps between the theory and the reality of the current technical implementation 😉